validphys2/src/validphys/filters.py

```diff
@@ -457,10 +506,20 @@ class Rule:
 
     numpy_functions = {"sqrt": np.sqrt, "log": np.log, "fabs": np.fabs}
 
-    def __init__(self, initial_data: dict, *, defaults: dict, theory_parameters: dict, loader=None):
+    def __init__(self, initial_data: dataclasses.dataclass, *, defaults: dict, theory_parameters: dict, loader=None):
```

Comment:
```suggestion
    def __init__(self, initial_data: FilterRule, *, defaults: dict, theory_parameters: dict, loader=None):
```

And I would be inclined to write in the error below that only `FilterRule`s are allowed.
The fact that we accept `Mapping`s is to make sure we don't break any new code yet but I don't think it should be advertised.